app/controllers/api/base_controller/normalizer.rb

```diff
@@ -19,9 +19,21 @@ def normalize_hash(type, obj, opts = {})
           end
         end
 
-        attrs.each do |k|
-          value = normalize_attr(k, obj.kind_of?(ActiveRecord::Base) ? obj.try(k) : obj[k])
-          result[k] = value unless value.nil?
+        case obj
```

Comment:
One of the biggest problems that remains to be solved is that certain models have already been serialized elsewhere before it even gets to this point, and arrive here as a hash. We need to address that so that we can eliminate most of the guesswork that's going on based only on attribute/key names